applications/GeoMechanicsApplication/tests/cpp_tests/test_geomechanics_erosion_piping.cpp

```diff
@@ -0,0 +1,211 @@
+    const auto& r_process_info = r_model_part.GetProcessInfo();
+
+    auto p_element_props = r_model_part.CreateNewProperties(0);
+    p_element_props->SetValue(CONSTITUTIVE_LAW, LinearElastic2DInterfaceLaw().Clone());
+
+    auto p_element = make_intrusive<UPwSmallStrainElement<2, 4>>(
+        0, CreateQuadrilateral2D4N(r_model_part, std::vector<int>{13, 14, 15, 16}, 3.0, 4.0, 2000.0, 2000.0),
+        p_element_props, std::make_unique<PlaneStrainStressState>());
+    p_element->Initialize(r_process_info);
+    r_model_part.AddElement(p_element);
+
+    // Set the pipe element properties
+    auto p_piping_element_prop = r_model_part.CreateNewProperties(1);
+    p_piping_element_prop->SetValue(PIPE_D_70, 2e-4);
+    p_piping_element_prop->SetValue(PIPE_ETA, 0.25);
+    p_piping_element_prop->SetValue(PIPE_THETA, 30);
+    p_piping_element_prop->SetValue(DENSITY_SOLID, 2650);
+    p_piping_element_prop->SetValue(DENSITY_WATER, 1000);
+    p_piping_element_prop->SetValue(PIPE_ELEMENT_LENGTH, 1.0);
+    p_piping_element_prop->SetValue(PIPE_MODIFIED_D, false);
+    p_piping_element_prop->SetValue(PIPE_MODEL_FACTOR, 1);
+    p_piping_element_prop->SetValue(PIPE_START_ELEMENT, 1);
+    p_piping_element_prop->SetValue(CONSTITUTIVE_LAW, LinearElastic2DInterfaceLaw().Clone());
+
+    // Create the start piping element
+    auto p_piping_element = make_intrusive<SteadyStatePwPipingElement<2, 4>>(
+        1, CreateQuadrilateral2D4N(r_model_part, std::vector<int>{1, 2, 3, 4}, 0.0, 1.0, 0.0, 500.0),
+        p_piping_element_prop, std::make_unique<PlaneStrainStressState>());
+    p_piping_element->Initialize(r_process_info);
+    r_model_part.AddElement(p_piping_element);
+
+    // Create other piping elements
+    p_piping_element = make_intrusive<SteadyStatePwPipingElement<2, 4>>(
+        2, CreateQuadrilateral2D4N(r_model_part, std::vector<int>{5, 6, 7, 8}, 2.0, 3.0, 500.0, 1000.0),
+        p_piping_element_prop, std::make_unique<PlaneStrainStressState>());
+    p_piping_element->Initialize(r_process_info);
+    r_model_part.AddElement(p_piping_element);
+
+    p_piping_element = make_intrusive<SteadyStatePwPipingElement<2, 4>>(
+        3, CreateQuadrilateral2D4N(r_model_part, std::vector<int>{9, 10, 11, 12}, 1.0, 2.0, 1000.0, 1500.0),
+        p_piping_element_prop, std::make_unique<PlaneStrainStressState>());
+    p_piping_element->Initialize(r_process_info);
+    r_model_part.AddElement(p_piping_element);
+
+    Parameters p_parameters(R"(
+    {
+ 		"max_piping_iterations": 25
+    }  )");
+
+    const auto p_solver =
+        Kratos::make_shared<SkylineLUFactorizationSolver<SparseSpaceType, LocalSpaceType>>();
+    const auto p_scheme =
+        Kratos::make_shared<BackwardEulerQuasistaticPwScheme<SparseSpaceType, LocalSpaceType>>();
+    const auto p_builder_and_solver =
+        Kratos::make_shared<ResidualBasedBlockBuilderAndSolver<SparseSpaceType, LocalSpaceType, KratosExecute::LinearSolverType>>(
```

Comment:
I believe the namespace qualification is redundant here, since the mock class is in namespace `Kratos`:
```suggestion
        make_shared<ResidualBasedBlockBuilderAndSolver<SparseSpaceType, LocalSpaceType, KratosExecute::LinearSolverType>>(
```

Reply:
Sorry, removing Kratos lead to error C2668 ambiguous call to overload function

Reply:
Apparently, it needs to stay there. Not sure why, but it seems I was wrong. Please ignore this suggestion.